test/intern/mixins/Tree.js

```diff
@@ -409,6 +409,34 @@ define([
 					}, null, 'Modification of child should not throw error');
 				});
 			});
+
+			test.test('child modification after parent when expanded', function() {
+				var dfd = this.async(10000);
+				expand(0).then(function() {
+					grid.collection.putSync({
+						id: '0',
+						value: 'Modified'
+					});
+					grid.collection.putSync({
+						id: '0:0',
+						value: 'Modified',
+						parent: '0'
+					});
+					setTimeout(dfd.callback(function() {
```

Comment:
Is there a reason this `setTimeout` needs to be here?  I think this test will work fine so long as you `return` on line 415... (though I mysteriously can't get it to fail for me on FF in either case with or without the fix)